### PR TITLE
Swaps: Fix some keyboard issues

### PR DESCRIPTION
**Pull request by jxom**
Fixes TEAM2-132
Fixes TEAM2-133

## What changed (plus any additional context for devs)

There were a few issues on Android where weird things would happen, such as:
- TEAM2-132: When you immediately close a sheet when it comes in focus, the input will refocus causing the sheet to reappear,
- TEAM2-133: When closing the swap settings sheet, it would go to an unexpected intermediate state

## PoW (screenshots / screen recordings)

https://www.loom.com/share/fdf7b369d9804380a160430d9fd8ece9

https://www.loom.com/share/d2a27e2a024c4ab19e0cf047b301cb51

### CODE REVIEW

File: src/hooks/useMagicAutofocus.js [hunk 1]
```diff
@@ -78,6 +93,10 @@ export default function useMagicAutofocus(
             delay = false;
           }, 200);
         });
+      } else if (!isNativeStackAvailable) {
+        // We need to do this in order to assure that the input gets focused
+        // when using fallback stacks.
+        InteractionManager.runAfterInteractions(fallbackRefocusLastInput);
```
Comment (jxom) [inline src/hooks/useMagicAutofocus.js:99]: moved the `fallbackRefocusLastInput` here as it was always calling `triggerFocus` twice.

**Review by estebanmino — COMMENTED** [review]
https://github.com/rainbow-me/rainbow/pull/3555#issuecomment-1162159033

**Review by derHowie — APPROVED** [review]
lgtm

**Comment by skylarbarrera** [comment]
https://user-images.githubusercontent.com/29204161/175187384-2f66855c-b4ff-427b-b544-3c2ae965d81a.mov

getting some weird behavior on swap settings when hiding keyboard or tapping done button

**Comment by estebanmino** [comment]
@skylarbarrera fixed in the last commit

https://user-images.githubusercontent.com/12115171/175342153-516820ed-c113-4dfd-b7e8-009e6d4328ff.mov

**Review by estebanmino — APPROVED** [review]
🚀 🔥 

EDIT: this was meant for other PR 😅 https://github.com/rainbow-me/rainbow/pull/3574#pullrequestreview-1017328350

**Pull request merged**